File: util/slug.rb
require "digest"

Encoding.default_external = 'utf-8'

salt = ARGV[0]
n = ARGV[1]

if Digest::SHA512.hexdigest(salt).slice(0, 8) != "a2ef2853" then
  puts "Input Error"
  exit
end

puts Digest::SHA512.hexdigest(salt + n).slice(0, 8)

File: util/slug.py
import sys, hashlib

if len(sys.argv) != 3:
  exit()

salt = sys.argv[1]
n = sys.argv[2]

if hashlib.sha512(salt.encode()).hexdigest()[0:8] != "a2ef2853":
  exit()

print(hashlib.sha512((salt + n).encode()).hexdigest()[0:8])
